.tekton/github-set-status-task.yaml

```
@@ -0,0 +1,38 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: github-set-status
+spec:
+  params:
+  - name: SHA
+  - name: STATE
+  volumes:
+  - name: githubtoken
+    secret:
+      secretName: githubtoken
+  steps:
+    - name: set-status
+      image: curlimages/curl:8.6.0
+      env:
+      - name: SHA
+        value: $(params.SHA)
+      - name: STATE
+        value: $(params.STATE)
+      volumeMounts:
+        - name: githubtoken
+          mountPath: /etc/github-set-status
+      script: |
+        #!/bin/sh
+        curl -L \
+             -X POST \
+             -H "Accept: application/vnd.github+json" \
+             -H "X-GitHub-Api-Version: 2022-11-28" \
+             -H "Authorization: Bearer $(cat /etc/github-set-status/token)" \
+             "https://api.github.com/repos/instana/python-sensor/statuses/${SHA}" \
+             -d "{
+                  \"state\":\"${STATE}\",
+                  \"target_url\":\"http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/default/pipelineruns/\",
+                  \"description\":\"Tekton build is in state: ${STATE}!\",
+                  \"context\":\"Tekton\"
+                 }"
```

For better readability, can we add `-H "Content-Type: application/json"` header to explicitly specify the content type as JSON and avoid adding escape characters?

```suggestion
        curl -L -X POST \
               -H "Accept: application/vnd.github+json" \
               -H "X-GitHub-Api-Version: 2022-11-28" \
               -H "Authorization: Bearer $(cat /etc/github-set-status/token)" \
               -H "Content-Type: application/json" \
               -d '{
                    "state": "'${STATE}'",
                    "target_url": "http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/default/pipelineruns/",
                    "description": "Tekton build is in state: '${STATE}'!",
                    "context": "Tekton"
                   }' \
              "https://api.github.com/repos/instana/python-sensor/statuses/${SHA}"

```

---

Valid point. I have also sorted the headers.